athenahealth/enum/appointment_status.go

```diff
@@ -0,0 +1,41 @@
+package enum
+
+import "strconv"
+
+// AppointmentStatus is derived from https://docs.athenahealth.com/api/api-ref/appointment#Book-appointment
+type AppointmentStatus string
+
+const (
+	Cancelled     AppointmentStatus = "x"
+	ChargeEntered AppointmentStatus = "4"
+	CheckedIn     AppointmentStatus = "2"
+	CheckedOut    AppointmentStatus = "3"
+	Future        AppointmentStatus = "f"
+	Open          AppointmentStatus = "o"
+)
+
+var appointmentStatusToString = map[AppointmentStatus]string{
+	Cancelled:     "x",
+	ChargeEntered: "4",
+	CheckedIn:     "2",
+	CheckedOut:    "3",
+	Future:        "f",
+	Open:          "o",
+}
+
+func (as AppointmentStatus) Int() int {
+	intVal, convErr := strconv.Atoi(appointmentStatusToString[as])
+	if convErr != nil {
+		return -1
+	}
+
+	return intVal
+}
+
+func (as AppointmentStatus) String() string {
+	return appointmentStatusToString[as]
+}
+
+func (as AppointmentStatus) Valid() bool {
+	return appointmentStatusToString[as] != ""
```

Comment:
`_, ok := map[val]` would be the preferred way to do this but I believe we'd prefer to have a slice of `AppointmentStatus`'s and iterate to check if `as` is included.

Reply:
I'll add them to a local variable in `Valid()`. I don't want to pollute the package autocomplete options with an array that just holds status values that are just used internally in one function